### PR TITLE
K8s cfg updates

### PR DESCRIPTION
**Pull request by jonhosmer**

### CODE REVIEW

File: deployments/production/spark-master-deployment.json [hunk 1]
```diff
@@ -3,32 +3,22 @@
     "kind": "Deployment",
     "metadata": {
         "labels": {
-            "component": "spark-cluster-master",
```
Comment (kalmanchapman) [inline deployments/production/spark-master-deployment.json:6]: will these changes (and the corresponding changes in the spark 2-1-0 deployment) change the name of the dns resolution `spark-cluster-master` -> `spark-master`? 

I believe these are ultimately positive changes, but we will need to update job files as well

File: deployments/production/spark-2-1-0-worker-daemonset.json [hunk 2]
```diff
@@ -35,12 +33,19 @@
...
+                                "name": "SPARK_WORKER_CORES",
+                                "valueFrom": {
+                                    "resourceFieldRef": {
+                                        "resource": "requests.cpu"
+                                    }
+                                }
+                            },
+                            {
                                 "name": "SPARK_MASTER_HOST",
                                 "value": "spark-cluster-master-2-1-0"
```
Comment (kalmanchapman) [inline deployments/production/spark-2-1-0-worker-daemonset.json:50]: per my other comment - I believe we will need to update these values `spark-cluster-master-2-1-0` -> `spark-master-2-1-0` if the dns resolution is changing

File: deployments/production/spark-2-1-0-master-deployment.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                "containers": [
+                    {
+                        "args": [
+                            "/bin/bash",
+                            "/usr/local/spark/sbin/start-master.sh"
+                        ],
+                        "env": [
+                            {
+                                "name": "SPARK_MASTER_HOST",
+                                "value": "spark-cluster-master-2-1-0"
```
Comment (kalmanchapman) [inline deployments/production/spark-2-1-0-master-deployment.json:33]: similar vein as other comments - not sure if we'll be able to listen here if dns doesn't resolve?

File: deployments/production/spark-worker-daemonset.json [hunk 4]
```diff
@@ -41,7 +39,19 @@
                             },
```
Comment (kalmanchapman) [inline deployments/production/spark-worker-daemonset.json:39]: ditto other comments - we'll need to update this if dns is resolving differently

**Review by kalmanchapman — COMMENTED** [review]
looks good to me - I think there might be some divergence between this pr and what's running on cluster re: pod labels `cluster-master` vs `-master`

File: deployments/production/spark-master-deployment.json (hunk 1 above)
Comment (jonhosmer) [reply deployments/production/spark-master-deployment.json:6]: This change specifically will not.  The labels only matter to the services selector (which I've also updated).  Changes to a service name _would_ impact the jobs, but those have not changed.  (I had thought about it, but reverted the service name changes due to the impact on job files, might revisit in the future though.)

File: deployments/production/spark-2-1-0-worker-daemonset.json (hunk 2 above)
Comment (jonhosmer) [reply deployments/production/spark-2-1-0-worker-daemonset.json:50]: Service name (dns hostname) has not changed (see my other comment).

File: deployments/production/spark-worker-daemonset.json (hunk 4 above)
Comment (jonhosmer) [reply deployments/production/spark-worker-daemonset.json:39]: Service name (dns hostname) has not changed (see my other comment).

**Comment by jonhosmer** [comment]
Everything here _should_ match exactly what is running in prod now (but let me know if you find something that isn't and we can update it).

**Pull request merged**